modules/openapi-generator/src/main/resources/dart2/class.mustache

```diff
@@ -190,22 +188,22 @@ class {{{classname}}} {
   static List<{{{classname}}}> listFromJson(List<dynamic> json, {bool emptyIsNull, bool growable,}) =>
     json == null || json.isEmpty
       ? true == emptyIsNull ? null : <{{{classname}}}>[]
-      : json.map((v) => {{{classname}}}.fromJson(v)).toList(growable: true == growable);
+      : (json as List<Map<String, dynamic>>).map({{{classname}}}.fromJson).toList(growable: true == growable);
 
   static Map<String, {{{classname}}}> mapFromJson(Map<String, dynamic> json) {
     final map = <String, {{{classname}}}>{};
-    if (json != null && json.isNotEmpty) {
-      json.forEach((String key, dynamic v) => map[key] = {{{classname}}}.fromJson(v));
+    if (json?.isNotEmpty == true) {
+      json.forEach((key, value) => map[key] = {{{classname}}}.fromJson(value as Map<String, dynamic>));
     }
     return map;
   }
 
   // maps a json object with a list of {{{classname}}}-objects as value to a dart map
   static Map<String, List<{{{classname}}}>> mapListFromJson(Map<String, dynamic> json, {bool emptyIsNull, bool growable,}) {
     final map = <String, List<{{{classname}}}>>{};
-    if (json != null && json.isNotEmpty) {
-      json.forEach((String key, dynamic v) {
-        map[key] = {{{classname}}}.listFromJson(v, emptyIsNull: emptyIsNull, growable: growable);
+    if (json?.isNotEmpty == true) {
+      json.forEach((key, value) {
+        map[key] = {{{classname}}}.listFromJson(value as List<dynamic>, emptyIsNull: emptyIsNull, growable: growable,);
```

Comment:
can you assess if this has any performance hit? Can casting things inside a loop slower be slower than casting upfront in loop definition?

Reply:
> can you assess if this has any performance hit?

I lack the knowledge for that...

> Can casting things inside a loop slower be slower than casting upfront in loop definition?

U mean to move this to the method signature? declaring `Map<String, dynamic> json` as `Map<String, List<dynamic>> json`?

On that note.. isn't this a microoptimization? I know that dart runtime will validate all assignment types whether you cast or not something, maybe the compiler make some optimizations when you cast before to not validate it again :thinking:... but again, I lack the knowledge for that.

oh, and [json_serializable](https://pub.dev/packages/json_serializable) also generates theses casts that I've added

Reply:
Casts usually don't have a runtime impact, they are mostly there to make the compiler happy.